src/nix/run.cc

```diff
@@ -13,6 +13,9 @@
 
 #if __linux__
 #include <sys/mount.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+#define pivot_root(new_root, put_old) (syscall(SYS_pivot_root, new_root, put_old))
```

Comment:
We already have definitions of `pivot_root` in `src/libstore/build/local-derivation-goal.cc` and `src/libstore/build/derivation-goal.cc` (which appears to be unused). Maybe we can move these into a header.

Reply:
Sure, I can move it if required. Any suggestions where to move to? I don't know what the conventions are. `src/libutil/util.hh` seems like a bit of a catch-all? But not sure if that's really better for such one-off defines.

Reply:
Dropped the macro instead. I think it just ends up being cruft too often. As you noted, 50% of the syscall macros in the entire code base are unused :grin:

I did make it consistent though, and there is already one place that calls `syscall(SYS_kill, ...)` for something platform specific on macos.